File: src/main/scala/org/xbery/artbeams/users/domain/Role.scala
package org.xbery.artbeams.users.domain

import org.xbery.artbeams.common.assets.domain.{Asset, AssetAttributes}

/**
  * Role entity.
  *
  * @author Radek Beran
  */
@SerialVersionUID(-1L)
case class Role(
  override val common: AssetAttributes,
  name: String
) extends Asset {
}

object Role {
  lazy val Empty = Role(
    AssetAttributes.Empty,
    ""
  )
}

File: src/main/kotlin/org/xbery/artbeams/users/domain/Role.kt
package org.xbery.artbeams.users.domain

import org.xbery.artbeams.common.assets.domain.Asset
import org.xbery.artbeams.common.assets.domain.AssetAttributes

/**
 * Role entity.
 *
 * @author Radek Beran
 */
data class Role(override val common: AssetAttributes, val name: String) : Asset() {
    companion object {
        val Empty: Role = Role(AssetAttributes.EMPTY, "")
    }
}
